### PR TITLE
loader-utils: concatenateChunksAsync perf increase

### PR DESCRIPTION
**Pull request by ibgreen**
Repeated concatenation of big chunks is very slow. Replace with a single concatenation at the end.

### CODE REVIEW

File: modules/loader-utils/src/lib/iterator-utils/async-iteration.js [hunk 1]
```diff
@@ -38,14 +38,23 @@ export async function forEach(iterator, visitor) {
  * This function can e.g. be used to enable atomic parsers to work on (async) iterator inputs
  */
 export async function concatenateChunksAsync(asyncIterator) {
-  let arrayBuffer = new ArrayBuffer(0);
-  let string = '';
+  const arrayBuffers = [new ArrayBuffer(0)];
+  const strings = [''];
```
Comment (isaacbrodsky) [inline modules/loader-utils/src/lib/iterator-utils/async-iteration.js:42]: why are these not just empty to start?

Comment (ibgreen) [reply modules/loader-utils/src/lib/iterator-utils/async-iteration.js:42]: Wanted to handle the case where the input iterator is empty and has no batches.

Comment (isaacbrodsky) [reply modules/loader-utils/src/lib/iterator-utils/async-iteration.js:42]: Could just create the array buffer at the end of the function then. This always concatenates an additional empty buffer.

Comment (ibgreen) [reply modules/loader-utils/src/lib/iterator-utils/async-iteration.js]: Yes you are right of course, updated. 

(I was so focused on optimizing this for 100 MB concatenations that I didn't give much attention to the perf in the simple cases.)

**Pull request merged**